src/test/java/org/opensearch/knn/index/OpenSearchIT.java

```diff
@@ -399,4 +399,71 @@ public void testExistsQuery() throws IOException {
 
         assertEquals(3, parseTotalSearchHits(EntityUtils.toString(response.getEntity())));
     }
+
+    public void testIndexingVectorValidation_updateVectorWithNull() throws Exception {
+        Settings settings = Settings.builder()
+                .put(getKNNDefaultIndexSettings())
+                .build();
+
+        createKnnIndex(INDEX_NAME, settings, createKnnIndexMapping(FIELD_NAME, 4));
+
+        // valid case with 4 dimension
+        final Float[] vectorForDocumentOne = {6.0f, 7.0f, 8.0f, 9.0f};
+        final String docOneId = "1";
+        addKnnDoc(INDEX_NAME, docOneId, FIELD_NAME, vectorForDocumentOne);
+
+        final Float[] vectorForDocumentTwo = {2.0f, 1.0f, 3.8f, 2.5f};
+        final String docTwoId = "2";
+        addKnnDoc(INDEX_NAME, docTwoId, FIELD_NAME, vectorForDocumentTwo);
+
+        //checking that one of two documents in retrievable based on closer similarity to query vector
+        int k = 1;
+        float[] queryVector = {5.0f, 6.0f, 7.0f, 10.0f};
+        final KNNQueryBuilder knnQueryBuilder = new KNNQueryBuilder(FIELD_NAME, queryVector, k);
+        final Response response = searchKNNIndex(INDEX_NAME, knnQueryBuilder, k);
+        final List<KNNResult> results = parseSearchResponse(EntityUtils.toString(response.getEntity()), FIELD_NAME);
+        assertEquals(1, results.size());
+        assertEquals(docOneId, results.get(0).getDocId());
+
+        //retrieving document by id
+        final Map<String, Object> knnDocMap = getKnnDoc(INDEX_NAME, docOneId);
+        final Float[] vectorInDocument = getVectorFromDocument(knnDocMap, FIELD_NAME);
+        assertEquals(vectorForDocumentOne.length, vectorInDocument.length);
+        assertArrayEquals(vectorForDocumentOne, vectorInDocument);
```

Comment:
Whats the purpose of this assertion? I think the test can remove this to simplify the test.

Reply:
Agree, we can assert documents fields after the update only, this particular assertion is redundant. I'll remove it in next commit